unit_tests/src/test_server_properties.cpp

```diff
@@ -29,3 +31,38 @@ TEST_CASE("server_properties", "[exceptions]")
                                  irods::KW_CFG_DB_PASSWORD));
     }
 }
+
+TEST_CASE("server_property_threadsafe", "[exceptions]")
+{
+    SECTION("Threads don't blow up")
+    {
+        auto& instance = irods::server_properties::instance();
+        const auto limit = 1000;
+
+        std::thread a([&]() {
+            std::string key;
+            for (auto i = 0; i < limit; i++) {
+                fmt::format_to(std::back_inserter(key), "{}", i);
+                instance.set_property(key, i);
+                key.clear();
+            }
```

Comment:
Why not use `std::to_string(i)` instead of `fmt::format_to`?

_This also applies to the other thread._